examples/heat_equation/source/heat_ida.cc

```diff
@@ -484,19 +492,29 @@ bool Heat<dim>::solver_should_restart (const double t,
                                           0,
                                           0,
                                           triangulation->locally_owned_subdomain());
+
+
       max_kelly = estimated_error_per_cell.linfty_norm();
       max_kelly = Utilities::MPI::sum(max_kelly, comm);
```

Comment:
this line is not very efficient since if we have many processes `max_kelly` will be much greater than what it really is. What do you suggest? 
```
max_kelly = Utilities::MPI::sum(max_kelly, comm) / Utilities::MPI::n_mpi_processes(comm)
```
?

@luca-heltai, @nicola-giuliani, @ESeNonFossiIo ?